Rare Spider Variants!

Armhulen opened the pull request:
:cl:
add: Rare Spider Variants have arrived! Every spiderling has a rare chance of growing up to a rare version of their original spider type!
add: Special thanks to Onule for the sprites, you're the best!
/:cl:


# SPIDERS HAVE BEEN GIVEN THE LOVE THEY DESERVE!

## New spider sprites!
NURSE =
![nurse](https://user-images.githubusercontent.com/17778097/30300091-afc79f66-9707-11e7-839e-9f6c9b851b29.png)
HUNTER =
![hunter](https://user-images.githubusercontent.com/17778097/30300102-bfff5da6-9707-11e7-96cf-a2bb16cb4ed1.png)
GUARD = 
![guard](https://user-images.githubusercontent.com/17778097/30300106-c4e9d15c-9707-11e7-9e8c-346271f9905a.png)
## Rare Spider Variants!
### Very rarely, a spiderling will become an evolved version of one of these creatures!
MIDWIFE = Gains a command function to message all spiders, can remotely kill off spiders. Praise the queen!
![nursenew](https://user-images.githubusercontent.com/17778097/30300220-500ea442-9708-11e7-839a-bf4f15bf6a7e.png)
VIPER = Lower health than the hunter and almost no melee damage but an **extremely** poisonous bite!
![hunternew](https://user-images.githubusercontent.com/17778097/30300232-6790d9a0-9708-11e7-993f-749f1e6a0359.png)
TARANTULA = Massive hulking spider that does crazy damage but moves slow off of webs. Great for protecting territory, will get shotgunned down outside of it.
![guardnew](https://user-images.githubusercontent.com/17778097/30300259-9aa7b0ac-9708-11e7-8419-fb543b5849bb.png)

Commit:
vipers use venom, tarantulas are fat and slow.

Commit:
viper!

added!

Commit:
tarantula!

done!

Armhulen commented:
ALSO PLANNED: Giving text explaining what each spider does so you don't get confused when you get a special variant

Ratvar commented:
Make vipers slowly generate poison for their attack, building over time? Make them inject acid into bloodstream? Fast 2hit(1hit, to be honest)death from speedy spider is not fun.
Queen-ability to announce things?
Max spider web per tile, so goliaths cannot spam one tile forever? Make webs collide into webby walls, so they can prevent fires and such, when 10 webs are on the same tile?

RemieRichards commented:
Nurse > Midwife
Giant > Tarantula

Armhulen commented:
i like it remie, i'll switch it

XDTM commented:
are spiders the new golems? count me in

Armhulen commented:
@PraiseRatvar the spider injects a toxin that will kill you EVENTUALLY if you don't go to medical, but it doesn't kill you any faster than a hunter. A hunter will actually kill you way faster with their 15 force attacks and 120 health.

@XDTM don't you dare >:(

XDTM commented:
Also, 80 poison is like ~7 minutes of 2 poison damage per tick, for a total of 400 poison damage per bite, if it's actually 80 units of reagent

Anonmare commented:
You could look up Paradise's Terror Spiders for inspiration
https://nanotrasen.se/wiki/index.php/Terror_Spider

DoctoreMad commented:
Daily reminder that just as 10 seconds is longer than you think and 7 minutes is shorter than you think

Armhulen commented:
@XDTM bring charcoal!

XDTM commented:
@DoctoreMad it's more than 10% of an average round

per bite

DoctoreMad commented:
Er, what does that mean?

XDTM commented:
counting an average round as 60 minutes, it's 7 minutes of poison per bite, aka a bit more of 10% of it

Armhulen commented:
you're basically immune to the effects if you have charcoal in your system though, i promise!

XDTM commented:
that's a best case scenario though, most won't have any

XDTM commented:
IMO it should be a normal amount of a stronger poison

Armhulen commented:
fighting spiders without charcoal is like fighting shadowlings without light? anyways i'll tone it down to 60

Armhulen commented:
i'd kinda like that but then i have to redo some of the var spider code and aaaaaaaaaaaaaaaaaaaaaauuuuuuuuuuuuuuuuuuuuuuuuuuuggggggggggggggggggghhhhhhhhhhh

XDTM commented:
not really, there's a var for the reagent to use as toxin, it's quite simple

Armhulen commented:
wait let me look at it again

Armhulen commented:
hoy fuck that's easy to change, any ideas for what toxin to use

XDTM commented:
there is the appropriately-named venom, lemme see how strong it is

XDTM commented:
power increases with the amount inside the target, seems ideal! Repeated bites means faster death, instead of a longer recovery period

Armhulen commented:
xtdm we're going places, i'll make that the venom

XDTM commented:
10-15 per bite should be balanced, if the bite itself doesn't do damage

XDTM commented:
by the way, move_to_delay only applies to ai movement, i'm pretty sure, which means that player tarantulas would wreck shit

Armhulen commented:
i think you're wrong? all the spiders had it before and they all move at different speeds.

XDTM commented:
`var/move_to_delay = 3 //delay for the automated movement.`

DoctoreMad commented:
Isn't move_to_delay the delay for idle(non-combat) movement?
(Note: noncoder)

Armhulen commented:
i see, let me test player speeds right now. also, how would you make it so you can't push a mob like juggernauts and goliaths

XDTM commented:
it's for AI-controlled movement

Armhulen commented:
>how would you make it so you can't push a mob like juggernauts and goliaths

XDTM commented:
lemme see;

meanwhile, set `speed = 3` (or higher/lower) to slow down the taratulas controlled by players

Armhulen commented:
okay thanks

XDTM commented:
to prevent pushing, set `status_flags = NONE`;
also add `mob_size = MOB_SIZE_LARGE`

Commit:
tarantulas are big guys

nobody pushin them round

Armhulen commented:
guess i'll start on the midwife, really good at making bases and can send messages to all spiders.

Armhulen commented:
also tell me if the tarantula values are too crazy.

Commit:
added maidwives, added spider channel

ExcessiveUseOfCobblestone commented:
sprites tho?

Commit:
me and cobby lose our minds over shitcode 2

electric boogaloo

Commit:
midwives have skins

shiny

Commit:
YES XHUIS GOT IT WORKING WHAT A GUY YEAAAAAAAAAAA

WOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

Armhulen commented:
i need sprites, i'll do them myself but i'm hoping someone comes along and does an amazing job

ExcessiveUseOfCobblestone commented:
the trick is to do your own then a spriter will get mad and make a good sprite for you.

XDTM commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,47 @@
+	poison_per_bite = 15
+	move_to_delay = 4
+	poison_type = "venom" //will soon after if there isn't medical attention!
+	speed = 4
```
Higher speed means slower movement, since it's actually the delay between steps

Armhulen commented:
aaaaah more things breaking one seconddddd

Jordie0608 commented:
But tarantulas are a small species

Armhulen commented:
i think it's the idea not the


i mean we don't have real spiders and the handwives aren't literally hand wives right

kevinz000 commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +336,36 @@
+	var/my_message
+	if(!message)
+		return
+	my_message = "<b>COMMAND FROM SPIDER QUEEN:</b> [message]
```
missing "

Armhulen replied:
i wouldn't review this, i just fixed a huge part of it don't worry

Armhulen replied:
i, as in cobby and oranges

Armhulen commented:
okay lets test this
i'm excited!!

Armhulen commented:
okay the code is working and it is working like a charm

now lets get to the cool part, new sprites

SkowronX commented:
>The spider can wiggle dance.

Armhulen commented:
> The spider can wiggle dance.

strut my stuff my stuff is so SHINY!

ShadowDeath6 commented:
I would not recommend venom since every time it metabolizes it has a 15% chance to decay into histamine(between 5-10 for 1.1 venom) which has an absurdly lethal overdose process regardless of time spent in the system

https://github.com/tgstation/tgstation/blob/master/code/modules/reagents/chemistry/reagents/toxin_reagents.dm#L406

Armhulen commented:
okay, any better ideas

Armhulen commented:
okay that's it, the new viper toxin is spewium

ShadowDeath6 commented:
possibly a weaker version of amanitin if you still want to go with the "kills you the longer it's in you" route

Armhulen commented:
>making a whole new toxin for a spider that has a 1/100 chance of spawning every time a spiderling grows up

ShadowDeath6 commented:
go ahead and use venom then, if it's that rare I doubt it'll be much of an issue anyway

Armhulen commented:
i'm probably going to use spewium for it's meme value

Armhulen commented:
nope nevermind stun combat sux

Commit:
new nurse sprite is in!

yay

Commit:
cosby joke about all my shit corrupting idk

this is gonna be annoying to fix

Commit:
Revert "cosby joke about all my shit corrupting idk"

This reverts commit 680b4c310e0b882f7465129c093e7f69fdff15c6.

Commit:
starts to add spidervenom, finishing tomorrow

Commit:
fuck god dammit shit fuck

Commit:
WE ARE BACK IN BIZNIZ TY ORANGES

Commit:
whoops

FantasticFwoosh commented:
With a bit more development, these could be a suitable replacement for xenomorphs like how paradise has terror spiders to fill in that role.

RemieRichards commented:
We dont need to replace anything you stupid fwoosh.

Bawhoppen commented:
I hate our spider sprite theme. Bay's spiders look much more menacing and cool.

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	health = 40
+	var/datum/action/innate/spider/comm/letmetalkpls
+
+	/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife/Initialize()
```
indentation

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	var/datum/action/innate/spider/comm/letmetalkpls
+
+	/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife/Initialize()
+		..()
```
. = ..()

Cyberboss commented on code/_globalvars/lists/mobs.dm:
```diff
@@ -20,6 +20,7 @@ GLOBAL_LIST_EMPTY(ai_list)
 GLOBAL_LIST_EMPTY(pai_list)
 GLOBAL_LIST_EMPTY(available_ai_shells)
 GLOBAL_LIST_EMPTY(simple_animals)
+GLOBAL_LIST_EMPTY(spidermobs)				//all spider mobs
```
these are only logged in spider mobs, fix the comment or change the functionality

Cyberboss replied:
preferably the latter

ExcessiveUseOfCobblestone replied:
Its only use is to message players [as of now at least] , it should only be Player-controlled spiders.

Cyberboss replied:
Yes, but it's much saner to keep track of all spiders and who cares if AI mobs get the message?

ExcessiveUseOfCobblestone replied:
fair enough

Armhulen replied:
i'll get around to this zzz

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +342,39 @@
+	return TRUE
+
+/datum/action/innate/spider/comm/proc/spider_command(mob/living/user, message)
+	var/my_message
```
why is this declared up here?

ExcessiveUseOfCobblestone replied:
he had some thing for deadchat last night where it was needed but I removed it without removing the var, whoops :p

Cyberboss replied:
Yeah, so move it down to there

ExcessiveUseOfCobblestone replied:
you are too smart for your own good cyberman!

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +342,39 @@
+	my_message = "<b>COMMAND FROM SPIDER QUEEN:</b> [message]"
+	for(var/mob/living/simple_animal/hostile/poison/giant_spider/M in GLOB.spidermobs)
+		to_chat(M, my_message)
+	log_talk(user, "SPIDERCOMMAND:[key_name(user)] : [message]",LOGSAY)
```
add a space after the first colon

Armhulen commented:
I'll fix all that when I have time, I'm probably going to go with 30u of polonium as it's slow acting but really deadly

DoctoreMad commented:
Instead of inserting just one toxin don't you think that you can add one base reagent that deals low toxin damage and another one that starts dealing damage after a amount of time passes?
It would also halve the time the poison stays in the target since charcoal will proc for both chemicals. At least that's how I think it works.

Armhulen commented:
Currently the code only allows for one kind of poison on the bite. I could MAKE a chemical that does that but it seems like a slog to do.

Commit:
feed me semour feed me all night long

fixes the cyberboss stuff except that complicated one i'll do something else with it for now

Commit:
temporary fix maybe

Armhulen commented:
Now hol up I'm getting sprites in and I need to add a spawn chance of these

XDTM commented:
Chemicals are pretty easy, also venom is still the right choice imo

FantasticFwoosh commented:
@RemieRichards watch how fast WJohn vanishes when xeno's are washed out of the code in favour of softer spider hive swarms.

Armhulen commented:
@XDTM I'll make a custom poison that is venom but without the decay into deathjuice

ShadowDeath6 commented:
https://github.com/tgstation/tgstation/pull/30593#issuecomment-328398410

:^)

Armhulen commented:
I'm not happy about it, I'm only doing it because it'll compile!

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -4,7 +4,7 @@
 #define SPINNING_COCOON 4
 
-/mob/living/simple_animal/hostile/poison
+/mob/living/simple_animal/hostile/poison //this guy's a fukkin idiot everyone knows that spiders inject venom not poison
```
It's a subtype of hostile mobs that transfer reagents and aren't exclusive to spiders.

See: Bees.

Armhulen replied:
hahahaha bees still inject venom haHA

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
 	poison_per_bite = 5
 	move_to_delay = 5
 
+//vipers are the rare variant of the hunter, no damage but so much poison medical care will be needed fast. Low hp, high speed and no damage so hit and run tactics.
```
They DO have damage though!

Armhulen replied:
one damage cmon now

ExcessiveUseOfCobblestone replied:
and rads/toxin damage...

Armhulen replied:
okay fine i'll reword it

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	melee_damage_upper = 1
+	poison_per_bite = 30
+	move_to_delay = 4
+	poison_type = "polonium" //slow acting but this at 30u per bite will seriously mess you up, perfect viper poison
```
This gives RADS

Armhulen replied:
yeah, rads are slow acting and do horrible effects if you don't get medical attention, not a terrible one for the viper

Armhulen replied:
i'm making a new one though

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+
+//tarantulas are really tanky, regenerating (maybe), hulky monster but are also extremely slow, so.
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula
+	desc = "Furry and black, it makes you shudder to look at it. This one has abyssal red eyes."
```
no name difference?

Armhulen replied:
deep > abyssal

Armhulen replied:
EYE COLOR is what i mean

ExcessiveUseOfCobblestone replied:
just like midwife you may want to add `name = "tarantula"`

Armhulen replied:
fine fine

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	icon_living = "guard"
+	icon_dead = "guard_dead"
+	maxHealth = 400 // woah nelly
+	health = 400
```
Just a heads up this is almost twice the health of a Juggernaut.

Armhulen replied:
200 health on the normal spider is already giant, i'll edit this down to 300 though good point

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	melee_damage_upper = 40
+	poison_per_bite = 0
+	move_to_delay = 8
+	speed = 3
```
This is not going to be fun to play as.

Armhulen replied:
i could make them fast on webs. oh dear me!

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,53 @@
+	mob_size = MOB_SIZE_LARGE
+
+//midwives are the queen of the spiders, can send messages to all them and web faster. That rare round where you get a queen spider and turn your 'for honor' players into 'r6siege' players will be a fun one.
+/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife
```
no name difference?

Armhulen replied:
no? midwife?

Armhulen replied:
i don't know what you mean here

ExcessiveUseOfCobblestone replied:
you didn't set a name var.

Armhulen replied:
why are we giving them special names? Oh whatever, it'll be kinda nice when it happens.

ExcessiveUseOfCobblestone replied:
You're the one giving them hierarchical structure, would make sense for the top dog to be a different name than the common peon.

ExcessiveUseOfCobblestone replied:
I wish I knew how to remove this entire section since it was changed :(

Armhulen replied:
i could change the pr then change it back that would close this right

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +342,39 @@
+	..()
+	GLOB.spidermobs += src
+
+/mob/living/simple_animal/hostile/poison/giant_spider/Logout(GLOB.spidermobs)
```
Remove this Arg

Armhulen replied:
caaaaan do

ExcessiveUseOfCobblestone reviewed (changes requested):
That Tarantula is not going to be fun to play as or against.

Armhulen commented:
whoops i should be fucking compiling lmao

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -445,6 +445,66 @@
+	toxpwr = 0
+
+/datum/reagent/toxin/spidervenom/on_mob_life(mob/living/M) //this is a big deal, the venom works like a virus kinda
+	var/stage = 1
```
You wrote this like a disease >.>

Armhulen replied:
is that bad?

Armhulen replied:
if it works ¯\_(ツ)_/¯

ExcessiveUseOfCobblestone replied:
needs to be a switch

Armhulen commented:
ultra wip on the venom, haha...

Armhulen commented:
also i just discovered switch so this code is about to get a lot cooler

Commit:
lets just have this for now and i'll fix this later

Commit:
all good

Commit:
fixes

Commit:
does what i should have done a long time ago

DoctoreMad commented:
> can remotely kill off spiders

HA HA TIME TO GRIEF is this logged

Armhulen commented:
One in 100, you could grief way easier with other spiders

Armhulen commented:
okay not with other spiders but with like, a maxcap. every round basically for that

Commit:
SKINS SKINS SKINS PULLAPALOOZA

Ratvar commented:
admin notify with rejenivate button every time spider queen kills someone?

WJohn commented:
These sprites kick ass!

ExcessiveUseOfCobblestone commented:
A changelog thanking onule would be really sweet.

Commit:
i think this does it okay

Commit:
spiders miright

Armhulen commented:
sounds good cobblestone

Commit:
merge conflict fix?

Armhulen commented:
I'll cut the remote execution for another pr, this is almost done!

Commit:
Merge remote-tracking branch 'tgstation/master' into spiders

yay?

Commit:
MISTAKES WERE MADE

Commit:
spider commands are cool now

Commit:
he protec

but he also attac

Commit:
ty joan

Commit:
ty joan again

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -103,6 +103,63 @@
+	status_flags = NONE
+	mob_size = MOB_SIZE_LARGE
+
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/Life(mob/living/M)
```
```
/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/movement_delay()
	var/turf/T = get_turf(src)
	if(locate(/obj/structure/spider/stickyweb) in T)
		speed = 2
	else
		speed = 7
	. = ..()
```

ChangelingRain commented on code/game/objects/effects/spiders.dm:
```diff
@@ -171,15 +180,18 @@
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider/tarantula, /mob/living/simple_animal/hostile/poison/giant_spider/hunter/viper, /mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife)
+				else
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider, /mob/living/simple_animal/hostile/poison/giant_spider/hunter, /mob/living/simple_animal/hostile/poison/giant_spider/nurse)
+				var/mob/living/simple_animal/hostile/poison/giant_spider/S = new grow_as(src.loc)
```
you've got some fucked indentation here; this and everything below it should all be tabbed down one

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +353,35 @@
 		stop_automated_movement = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
```
. = ..()

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -295,6 +353,35 @@
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
+	GLOB.spidermobs += src
```
add this somewhere and make this line `GLOB.spidermobs[src] = TRUE`
```
/mob/living/simple_animal/hostile/poison/giant_spider/Destroy()
	GLOB.spidermobs -= src
	return ..()
```
